Pokedex/SearchViewController.swift

```diff
@@ -0,0 +1,335 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        shouldShowSearchResults = false
+      
+        // Do any additional setup after loading the view.
+        setupTableView()
+        setupSearch()
+    }
+
+    func setupSearch(){
+        searchController = UISearchController(searchResultsController: nil)
+        //searchController.searchBar.frame = CGRect(x: 0.0, y: 0.0, width: view.frame.width, height: 44)
+
+        searchController.searchResultsUpdater = self
+        searchController.dimsBackgroundDuringPresentation = false
+        searchController.searchBar.placeholder = "Search Pokemon"
+        searchController.hidesNavigationBarDuringPresentation = false
+        searchController.searchBar.delegate = self
+
+        navigationItem.titleView = searchController.searchBar
+        definesPresentationContext = true
+
+        
+        
+        searchController.searchBar.isHidden = false
+        searchController.searchBar.sizeToFit()
+        // This following line fixed a  bug for me where my search bar would not display....
+        searchController.searchBar.scopeButtonTitles = []
+
+        // Place the search bar view to the tableview headerview.
+        tableView.tableHeaderView = searchController.searchBar
+        
+
+    }
+    
+    func setupTableView(){
+        tableView.frame = CGRect(x: 0, y: UIApplication.shared.statusBarFrame.height +
+            self.navigationController!.navigationBar.frame.height, width: view.frame.width, height: view.frame.height)
+        tableView = UITableView(frame: tableView.frame)
+        tableView.contentInset = UIEdgeInsetsMake(0, 0, self.bottomLayoutGuide.length + 150, 0)  //this is to prevent tab bar from hiding last cell
+        
+        tableView.rowHeight = 80
+        tableView.backgroundColor = UIColor.white
+        tableView.register(PokemonTableViewCell.self, forCellReuseIdentifier: "pokemonCell")
+        tableView.register(CategoryTableViewCell.self, forCellReuseIdentifier: "categoryCell")
+        tableView.delegate = self
+        tableView.dataSource = self
+        view.addSubview(tableView!)
+    }
+    
+
+    
+//    func setupTypeTableView(){
+//        typeTableView.frame = CGRect(x: 0, y: 0, width: view.frame.width, height: view.frame.height)
+//        typeTableView = UITableView(frame: typeTableView.frame)
+//        typeTableView.contentInset = UIEdgeInsetsMake(0, 0, self.bottomLayoutGuide.length + 150, 0)
+//        typeTableView.delegate = self
+//        typeTableView.dataSource = self
+//        typeTableView.backgroundColor = UIColor.white
+//        typeTableView.register(PokemonTableViewCell.self, forCellReuseIdentifier: "pokemonCell")
+//        view.addSubview(typeTableView!)
+//    }
+//    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        viewWillAppear(false)
+    }
+}
+
+extension SearchViewController: UITableViewDelegate, UITableViewDataSource{
+    // MARK: - Table View
```

Comment:
your search view controller is not set up correctly. For one, you didn't need to make another table view for all the options of search. Could have made labels and text boxes that received inputs and then based on those inputs and after a user clicks a search button, you load a table view of pokemon that meet their criterion. You already did the work for loading a table view in your featured tab, just had to pass in the right pokemon to load the data.